npm/extensions.bzl

```diff
@@ -63,7 +65,23 @@ def _extension_impl(module_ctx):
             if not attr.pnpm_lock:
                 continue
 
-            lock_importers, lock_packages, lock_patched_dependencies = utils.parse_pnpm_lock(module_ctx.read(attr.pnpm_lock))
+            if attr.use_starlark_yaml_parser:
```

Comment:
maybe this should WARN since we'd like to drop the yaml parser code eventually?

Reply:
given that some users will be force into the yaml parser due to https://github.com/pnpm/pnpm/issues/5414, I'd rather wait on the warning until that issue is resolved and the warning can suggest to upgrade to the latest pnpm version.